src/zcash/Address.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Address.hpp"
+#include "NoteEncryption.hpp"
+#include "prf.h"
+
+namespace libzcash {
+
+uint256 ViewingKey::pk_enc() {
+    return ZCNoteEncryption::generate_pubkey(*this);
+}
+
+ViewingKey SpendingKey::viewing_key() {
+    return ViewingKey(ZCNoteEncryption::generate_privkey(*this));
+}
+
+SpendingKey SpendingKey::random() {
+    return SpendingKey(random_uint256());
```

Comment:
Eventually I want `SpendingKey` to have its 4 leading bits 0, and then the PRF to check that those bits are 0, instead of just masking them off.